packages/xsnap/test/test-xsnap.js

```diff
@@ -302,6 +303,44 @@ const writeAndReadSnapshot = async (t, snapshotUseFs) => {
 test('write and read snapshot (use FS)', writeAndReadSnapshot, true);
 test('write and read snapshot (use stream)', writeAndReadSnapshot, false);
 
+test('produce golden snapshot hashes', async t => {
+  const toEvals = [
+    [`no evaluations`, ''],
+    [
+      `smallish safeInteger multiplication doesn't spill to XS_NUMBER_KIND`,
+      `globalThis.bazinga = 100; globalThis.bazinga *= 1_000_000;`,
+    ],
+  ];
+  for await (const [description, toEval] of toEvals) {
+    t.log(description);
+    const messages = [];
+    async function handleCommand(message) {
+      messages.push(decode(message));
+      return new Uint8Array();
+    }
+
+    const vat0 = await xsnap({
+      ...options(io),
+      handleCommand,
+      snapshotUseFs: false,
+    });
+    if (toEval) {
+      await vat0.evaluate(toEval);
+    }
+
+    const hash = createHash('sha256');
+    for await (const buf of vat0.makeSnapshotStream()) {
+      hash.update(buf);
+    }
+    await vat0.close();
+
+    const hexHash = hash.digest('hex');
+    t.log(hexHash);
+    t.snapshot(hexHash, description);
```

Comment:
Let's add a note here to remind future test-runners of what we're asserting.

This test snapshot was created from a version of xsnap that was manually observed to avoid the spill (because it has the `__builtin` and can safely tell when it remains within the safe range of integers). We cannot test for the spill/not-spill directly without either parsing the `xsnap` heap snapshot, or writing C test code that performs a multiplication and then inspects the `.kind` field of the returned value. But we know that any change to the spill/non-spill behavior *will* change the heap snapshot hash.

The snapshot will change (and the test will fail) if either:
* XS is changed in a way that changes the spill behavior (e.g. maybe they give up on that optimization for some reason)
* XS is changed in any other way that changes the heap snapshot

It probably can't change in response to the compiler having/not-having the builtins, because the `-D` change means the not-having-builtins cases will simply fail to link, well before we get to this test case. And it won't change in response to the compiler having/not-having `__has_builtin()`, because the `-D` means we completely ignore the normal `__has_builtin()`.

So as automated tests go, it's a bit too sensitive to things that 1: will happen, eventually, and 2: aren't what we're trying to assert. That's fine, especially since checking the real thing is too hard, but we should add a note, so when the test fails and the developer is considering ava's `--update-snapshots`, they can tell why this test was using a snapshot in the first place.